data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java

```diff
@@ -59,6 +61,29 @@ public ChainDataProvider(
     this.chainStorageClient = chainStorageClient;
   }
 
+  public static ValidatorStatus getValidatorStatus(Validator validator, UnsignedLong epoch) {
+    if (epoch.compareTo(validator.getActivation_eligibility_epoch()) < 0) {
+      return ValidatorStatus.DEPOSITED;
+    }
+    if (epoch.compareTo(validator.getActivation_epoch()) < 0) {
+      return ValidatorStatus.PENDING;
+    }
+    if (validator.getExit_epoch().equals(Constants.FAR_FUTURE_EPOCH)) {
+      return ValidatorStatus.ACTIVE;
+    }
+    if (epoch.compareTo(validator.getExit_epoch()) < 0) {
+      if (validator.isSlashed()) {
+        return ValidatorStatus.SLASHING;
+      } else {
+        return ValidatorStatus.EXITING;
```

Comment:
i'd be super tempted to use a ternary here
```
return validator.isSlashed() ? SLASHING : EXITING;
```

Reply:
done